libcst/_nodes/statement.py

```diff
@@ -948,6 +1024,75 @@ def _codegen_impl(self, state: CodegenState) -> None:
                 finalbody._codegen(state)
 
 
+@add_slots
+@dataclass(frozen=True)
+class TryStar(BaseCompoundStatement):
+    """
+    A ``try`` statement with ``except*`` blocks.
+    """
+
+    #: The suite that is wrapped with a try statement.
+    body: BaseSuite
+
+    #: A list of one or more exception handlers.
+    handlers: Sequence[ExceptStarHandler]
+
+    #: An optional else case.
+    orelse: Optional[Else] = None
+
+    #: An optional finally case.
+    finalbody: Optional[Finally] = None
+
+    #: Sequence of empty lines appearing before this compound statement line.
+    leading_lines: Sequence[EmptyLine] = ()
+
+    #: The whitespace that appears after the ``try`` keyword but before
+    #: the colon.
+    whitespace_before_colon: SimpleWhitespace = SimpleWhitespace.field("")
+
+    def _validate(self) -> None:
+        if len(self.handlers) == 0:
+            raise CSTValidationError(
+                "A TryStar statement must have at least one ExceptHandler"
+            )
+
+    def _visit_and_replace_children(self, visitor: CSTVisitorT) -> "TryStar":
+        return TryStar(
+            leading_lines=visit_sequence(
+                self, "leading_lines", self.leading_lines, visitor
+            ),
+            whitespace_before_colon=visit_required(
+                self, "whitespace_before_colon", self.whitespace_before_colon, visitor
+            ),
+            body=visit_required(self, "body", self.body, visitor),
+            handlers=visit_sequence(self, "handlers", self.handlers, visitor),
+            orelse=visit_optional(self, "orelse", self.orelse, visitor),
+            finalbody=visit_optional(self, "finalbody", self.finalbody, visitor),
+        )
+
+    def _codegen_impl(self, state: CodegenState) -> None:
+        for ll in self.leading_lines:
+            ll._codegen(state)
+        state.add_indent_tokens()
+
+        end_node = self.handlers[-1]
+        orelse = self.orelse
+        end_node = end_node if orelse is None else orelse
+        finalbody = self.finalbody
+        end_node = end_node if finalbody is None else finalbody
```

Comment:
I can see that this is consistent with existing style, and I approve of it in terms of this one diff, we ought to respect the style elsewhere.

But do we really think overwriting `end_node` 3 times is clearer than having an if/else block that just sets it to the appropriate value? I find it confusing, and I'm pretty sure it's also less efficient to do everything in backward order like this! I'd be happy to put up a PR trying to clean this up a bit in the whole module.

Reply:
I also found it confusing, feel free to put up that PR!